monai/bundle/scripts.py

```diff
@@ -477,34 +491,52 @@ def load(
             progress=progress,
             args_file=args_file,
         )
-        train_config_file = bundle_dir_ / name / "configs" / f"{workflow_type}.json"
-        if train_config_file.is_file():
-            _net_override = {f"network_def#{key}": value for key, value in net_override.items()}
-            _workflow = create_workflow(
-                workflow_name=workflow_name,
-                args_file=args_file,
-                config_file=str(train_config_file),
-                workflow_type=workflow_type,
-                **_net_override,
-            )
-        else:
-            _workflow = None
 
     # loading with `torch.jit.load`
     if load_ts_module is True:
         return load_net_with_metadata(full_path, map_location=torch.device(device), more_extra_files=config_files)
     # loading with `torch.load`
     model_dict = torch.load(full_path, map_location=torch.device(device))
+
     if not isinstance(model_dict, Mapping):
         warnings.warn(f"the state dictionary from {full_path} should be a dictionary but got {type(model_dict)}.")
         model_dict = get_state_dict(model_dict)
 
-    if model is None and _workflow is None:
+    if return_state_dict:
         return model_dict
-    model = _workflow.network_def if model is None else model
-    model.to(device)
 
-    copy_model_state(dst=model, src=model_dict if key_in_ckpt is None else model_dict[key_in_ckpt], **copy_model_args)
+    _workflow = None
+    if model is None and net_name is None:
+        bundle_config_file = bundle_dir_ / name / "configs" / f"{workflow_type}.json"
+        if bundle_config_file.is_file():
+            _net_override = {f"network_def#{key}": value for key, value in net_override.items()}
+            _workflow = create_workflow(
+                workflow_name=workflow_name,
+                args_file=args_file,
+                config_file=str(bundle_config_file),
+                workflow_type=workflow_type,
+                **_net_override,
+            )
+        else:
+            warnings.warn(f"Cannot find the config file: {bundle_config_file}, return state dict instead.")
+            return model_dict
+        if _workflow is not None:
+            if getattr(_workflow, "network_def") is None:
```

Comment:
```suggestion
            if not hasattr(_workflow, "network_def"):
```
Using `getattr` with no default value will raise an exception, I don't think that was intended here?

Reply:
Yeah, thanks!